### ISSUE TITLE
Agents 

### ISSUE THREAD

**Issue by OskarStark** [php-llm/llm-chain#177]
Do you think it could make sense to create an agent which uses different configured chains. This could also be nicely configured via the bundle.

**Comment by chr-hertel** [comment]
Yes, that's def one approach - having the chain as tool - duplicate of #30 tho

**Comment by chr-hertel** [comment]
> I really like the examples of this Autogen course, but I guess this is a huge effort - not sure what the first step would be and if that would rather be an additional lib on top of this one:
> 
> https://learn.deeplearning.ai/courses/ai-agentic-design-patterns-with-autogen

Closing #30 in favor of this one.

**Comment by chr-hertel** [comment]
So, with 0.19 chain in chain is possible 💪 but that's only using leveraging tool calls.

What I'm missing: interaction with a subchain: what if the subchain has a question to the user?
